client/landing/stepper/declarative-flow/internals/steps-repository/create-site/index.tsx

```diff
@@ -144,10 +144,7 @@ const CreateSite: Step = function CreateSite( { navigation, flow, data } ) {
 	);
 	const urlQueryParams = useQuery();
 	const skipMigration = urlQueryParams.get( 'skipMigration' ) || '';
-	const useThemeHeadstart =
-		! isStartWritingFlow( flow ) &&
-		! isNewHostedSiteCreationFlow( flow ) &&
-		! isMigrationSignupFlow( flow );
+	const useThemeHeadstart = ! isStartWritingFlow( flow ) && ! isNewHostedSiteCreationFlow( flow );
```

Comment:
Follow-up concern: we should probably disable headstart for sites created from the hosted-site-migration flow.

Reply:
Created this issue to tackle follow up cleanup: https://github.com/Automattic/wp-calypso/issues/101586